server/raft.go

```diff
@@ -3619,6 +3613,20 @@ func (n *raft) processAppendEntryResponse(ar *appendEntryResponse) {
 		n.stepdownLocked(noLeader)
 		n.Unlock()
 		arPool.Put(ar)
+		return
```

Comment:
Is it right to `return` in this case? Looks like we will step down but won't actually process the AE we were sent.

Reply:
I don't think we should process the response of a node that is ahead of us. They have another leader (since we're behind), so the request we sent should be ignored.